pkg/scheduler/scheduler.go

```diff
@@ -466,18 +471,20 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 				} else {
 					klog.V(5).InfoS("Status after running PostFilter plugins for pod", "pod", klog.KObj(pod), "status", status)
 				}
-				if status.IsSuccess() && result != nil {
-					nominatedNode = result.NominatedNodeName
+				if result != nil {
+					nominatedNode = &result.NominatedNodeName
 				}
 			}
 			// Pod did not fit anywhere, so it is counted as a failure. If preemption
 			// succeeds, the pod should get counted as a success the next time we try to
 			// schedule it. (hopefully)
 			metrics.PodUnschedulable(fwk.ProfileName(), metrics.SinceInSeconds(start))
 		} else if err == ErrNoNodesAvailable {
+			nominatedNode = pointer.StringPtr("")
 			// No nodes available is counted as unschedulable rather than an error.
 			metrics.PodUnschedulable(fwk.ProfileName(), metrics.SinceInSeconds(start))
 		} else {
+			nominatedNode = pointer.StringPtr("")
```

Comment:
in theory, this is the case of a transient error. Perhaps we shouldn't clear the nominated node here?

Reply:
It's a bit debatable depends how we interpret this kind of error. If we believe it can be auto-healed, we shouldn't clear it; otherwise clear it to prevent the pod from occupying resources.

I don't have a strong option as personally, I don't have real-world data points to determine if it's transient and how transient it is... I'm giving a pessimistic judgment here :(

Reply:
I also don't have a strong opinion. But I think that if the pod really needs preemption, it would get a nominated node in the next scheduling attempt. So it seems safer for utilization to clear.